hs_core/tests/api/native/test_create_new_version_resource.py

```diff
@@ -0,0 +1,198 @@
+
+        # test to make sure a new unique identifier has been created for the new versioned resource
+        self.assertIsNotNone(new_res_generic.short_id, msg='Unique identifier has not been created for new versioned resource.')
+        self.assertNotEqual(new_res_generic.short_id, self.res_generic.short_id)
+
+        # test to make sure the new versioned resource has no content files
+        self.assertEqual(new_res_generic.files.all().count(), 0)
+        # test to make sure the new versioned resource has the correct identifier
+        self.assertEqual(new_res_generic.metadata.identifiers.all().count(), 1, msg="Number of identifier elements not equal to 1.")
+        self.assertIn('hydroShareIdentifier', [id.name for id in new_res_generic.metadata.identifiers.all()],
+                      msg="hydroShareIdentifier name was not found for new versioned resource.")
+        id_url = '{}/resource/{}'.format(hydroshare.utils.current_site_url(), new_res_generic.short_id)
+        self.assertIn(id_url, [id.url for id in new_res_generic.metadata.identifiers.all()],
+                      msg="Identifier url was not found for new versioned resource.")
+
+        # test to make sure the new versioned resource is linked with the original resource via isReplacedBy and isVersionOf metadata elements
+        self.assertGreater(new_res_generic.metadata.relations.all().count(), 0, msg="New versioned resource does has relation element.")
+        self.assertIn('isVersionOf', [rel.type for rel in new_res_generic.metadata.relations.all()],
+                      msg="No relation element of type 'isVersionOf' for new versioned resource")
+        version_value = '{}/resource/{}'.format(hydroshare.utils.current_site_url(), self.res_generic.short_id)
+        self.assertIn(version_value, [rel.value for rel in new_res_generic.metadata.relations.all()],
+                      msg="The original resource identifier is not set as value for isVersionOf for new versioned resource.")
+        self.assertIn('isReplacedBy', [rel.type for rel in self.res_generic.metadata.relations.all()],
+                      msg="No relation element of type 'isReplacedBy' for the original resource")
+        version_value = '{}/resource/{}'.format(hydroshare.utils.current_site_url(), new_res_generic.short_id)
+        self.assertIn(version_value, [rel.value for rel in self.res_generic.metadata.relations.all()],
+                      msg="The new versioned resource identifier is not set as value for isReplacedBy for original resource.")
+
+        # test isReplacedBy is removed after the new versioned resource is deleted
+        hydroshare.delete_resource(new_res_generic.short_id)
+        self.assertNotIn('isReplacedBy', [rel.type for rel in self.res_generic.metadata.relations.all()],
+                         msg="isReplacedBy is not removed from the original resource after its versioned resource is deleted")
+
+
+    def test_new_version_raster_resource(self):
```

Comment:
@pkdash @ironfroggy I thought about this when I wrote the tests, and agree with @ironfroggy we should not need write tests for each specific resource type since we use a generic function to handle all (resource contents and metadata) at the hs_core level. Current test includes a representative raster resource type testing, which should be sufficient. I do agree it'd be nice though to test each resource type just to double make sure, but it'd be probably better for each resource type to test it individually rather than at hs_core level inside this test, since when new resource type is added, it is up to individual resource types to decide whether they want to add this resource-versioning test for their resource type or not, at the hs_core level, I think these two test functions should be sufficient. Thanks for bringing this up.